src/facility/facility.service.ts

```diff
@@ -1,158 +1,174 @@
-            }
-            if(facility.photos)
-            {
-                facility.photos=data.photos
-            }
-            await this.facilityRepository.save(facility);
-            const {...result} = facility
-            return {
-                success:true,
-                data: result
-            };
-        }
-        else {
-            return {
-                sucess:false,
-                message: "Updatation failed"
-            }
-        }
-    }
-    async searchDistrict(facility1:Facility,data:any): Promise<any> {
-        const district = data.district
-        const facility = await this.facilityRepository.find({district});
-        if(facility) {
-            const {...result}=facility
-            return {
-                success:true,
-                data: result
-            };
-        }
-        else {
-            return {
-                success: false,
-                message : "no such facility exists"
-            }
-        }
-
-    }
-    //get all districts
-    async getDistricts():Promise<any>{
-        return await this.facilityRepository.getDistricts();
-    }
-}
+import { Injectable,Logger, HttpException, HttpStatus, ParseIntPipe} from '@nestjs/common';
+import { FacilityRepository } from './facility.repository'; 
+import { InjectRepository } from '@nestjs/typeorm';
+import { Facility } from './entities/Facility.entity';
+import { UpdateFacilityDto } from './dto';
+import { User } from 'src/auth/entities/User.entity';
+import { UserRepository } from 'src/auth/user.repository';
+
+@Injectable()
+export class FacilityService {
+    private logger = new Logger('Facility Controller');
+    constructor(
+   @InjectRepository(FacilityRepository)
+   @InjectRepository(UserRepository)
+   private readonly facilityRepository: FacilityRepository,
+   private readonly userRepository:UserRepository){}
+   gethello(): string { 
+        return 'Welcome to stay service';
+    }
+    
+
+    async addfacility(data:any,user1:User): Promise<any> {
+            const user = await this.userRepository.findOne({id:user1.id})
+            this.logger.verbose(user.type)
+            if(user.type === 'facilityowner'){
+                data.status = 'ACTIVE';
+                data.ownerID=user.id;
+                const registerStay = await this.facilityRepository.save(data);
+                const {...result } = registerStay;
+                return{
+                    success: true,
+                    message: 'Success',
+                    data: result,
+                }
+            }
+            else{
+                throw new HttpException("Action Forbidden",HttpStatus.FORBIDDEN);
+            }
+        
+        
+    }
+
+    async getAllFacility(): Promise<any> {
+        return this.facilityRepository.getAllFacility();
+    }
+   
+
+    async getFacility(user1:User): Promise<any> {
+        const id = user1.id
+        const user = await this.userRepository.findOne({id:user1.id})

```

Comment:
cross check invocations , we should be getting a user object from request.

Reply:
done